### DISCUSSION TITLE
AI Assistant enabled in Community version

### DISCUSSION THREAD

**Discussion by kevgeni** [dbeaver/dbeaver#26332] [Q&A] [answered]
### Description

In community version 24.0.3
https://dbeaver.io/2024/04/21/dbeaver-24-0-3/
we have this

AI:
* AI assistant is enabled by default
* Database metadata collection was fixed

There bug here is that 
* There is no way to disable the AI aintegration since the ai settings are not available in the community edition
  * https://dbeaver.com/docs/dbeaver/AI-Smart-Assistance/
* Enabling ai and Database Metadata collection and sent that to whatever external system
*    enabling by default, opt-in
*    without a user consent is bad, really bad. 
*    to me, it's a security issue to have dbeaver auto export my data structure. 

I can not export my database structure to any existing ai/gpt, I view this as security issue. 
I can not find how to disable this ai integration. I can not find any documentation about what "ai assistant" do.
I may need to stop using dbeaver if this ai is enabled by default and cannot be disabled. 

Please advise. 

### DBeaver Version

24.0.3

**Comment by Destrolaric** [comment, marked as answer]
@kevgeni Greetings. This mention does not impact users without the AI plugin installed. In the Community edition, AI is not installed by default. For users who have not installed the AI plugin separately, no AI is used, and metadata is not collected.

**Comment by kevgeni** [comment]
Thank you very much @Destrolaric for your quick answer. Can you please explain why then it's written as "Enabled by default" in the changelog page of dbeaver community? If ai plugin is not installed by default, how can the "AI assistant is enabled by default" and how can "Database metadata collection was fixed"/is also enabled?

**Comment by E1izabeth** [comment]
AI assistant was enabled by default in PRO products and in Community Edition only if the plugin installed. We probably need to write more detailed release notes.

Also, even if someone installed this plugin, here is an option to turn it off
![image](https://github.com/dbeaver/dbeaver/assets/28875055/0e041e06-09ae-4d3d-9396-33fc3f56081d)

Besides that, we don't send any data without confirmation. When the user press the button to use it for the first time, we show a confirmation
![image](https://github.com/dbeaver/dbeaver/assets/28875055/40c61204-30ab-4fd9-88de-cb150d552650)
And then it works only if you put some text in the natural language and press `Translate` button
![image](https://github.com/dbeaver/dbeaver/assets/28875055/49a8bffe-6a96-41dd-92f3-a14166abe532)

So, there is no way to acidentially use it.

**Comment by Destrolaric** [comment]
`Database metadata collection was fixed` - yep, this is too short. I will specify what exactly was fixed. When the user wants to query AI, he can specify which metadata about the database is needed for the prompt. If there is a lot of metadata, AI will not be able to process the query. We fixed our prompt truncate functionality so the query will not fail due to that.

**Comment by kevgeni** [comment]
Thank you very much for your clarification. I understand that I will not be impacted by this changes since I will not install the ai plugin in my community version. 
I do find the idea to auto enable this... maybe not a good one, since I don't know what it means. 
if I execute a sql command and it auto ai it, it's bad, but if I need to volontarily open an ai prompt, it's probably ok. please excuse my lack of knowledge in this part.


as long as you improve the documentation concerning the community changelog and the documentation regarding the ai plugin to install in community, and that it's not installed by default, then I am happy

**Comment by Destrolaric** [comment]
`volontarily open an ai prompt,`
Yep, AI is called only during manual prompt in CE or a special Chat mode in PRO editions.

**Comment by thanghust** [comment]
I download the Zip file version then it doesn't show the AI feature, but the Installer version does have. Any one tell me why ?

**Reply by lauma** [reply]
I installed DBeaver Community around month ago (windows installation) and I also had installed and enabled AI plugins automatically. I wish I were asked beforehand and could refuse, otherwise why it is called a plugin then...
